src/containerapp-preview/azext_containerapp_preview/_utils.py

```diff
@@ -0,0 +1,90 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.azclierror import NoTTYError
+from azure.cli.core.extension.dynamic_install import _get_extension_use_dynamic_install_config
+from knack.prompting import prompt_y_n, NoTTYException
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+MIN_GA_VERSION = '0.3.35'
+
+
+def _get_or_add_extension(cmd, extension_name):
+    from azure.cli.core.extension import (
+        ExtensionNotInstalledException, get_extension)
+    from pkg_resources import parse_version
+    prompt_ext = True
+    try:
+        ext = get_extension(extension_name)
+        # check extension version
+        if ext and parse_version(ext.version) < parse_version(MIN_GA_VERSION):
+            msg = f"The command requires the latest version of extension {extension_name}. Run 'az extension add --upgrade -n {extension_name}' to upgrade extension"
+            logger.warning(msg)
+            return False
+    except ExtensionNotInstalledException:
+        prompt_msg = f"The command requires the extension {extension_name}. Do you want to install it now?"
+        prompt_ext = _prompt_y_n(cmd, prompt_msg, extension_name)
```

Comment:
Suggest renaming this method as it can easily confuse with method `prompt_y_n` from knack which affect readability.